tsdb/wlog/watcher.go

```diff
@@ -607,12 +621,12 @@ func (w *Watcher) readSegment(r *LiveReader, segmentNum int, tail bool) error {
 			if !w.sendMetadata {
 				break
 			}
-			metadata, err = dec.Metadata(rec, metadata[:0])
+			meta, err := dec.Metadata(rec, metadata[:0])
```

Comment:
This is no longer re-using the memory.  I can't see why this was changed.
As it stands `metadata` is always `nil` so you could remove it as a variable.